File: frontend/src/scripts/fix_home.py

import os

file_path = r"c:\S\Spensmer\frontend\src\pages\public\Home.js"

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Lines to delete: 708 to 849 (1-based) => indices 707 to 848
# However, let's verify content to be safe.
# Line 708 should be "</motion.div>"
# Line 849 should be "</motion.div>"

start_idx = 707
end_idx = 849

if lines[start_idx].strip() == "</motion.div>" and lines[end_idx-1].strip() == "</motion.div>":
    print("Found exact block to delete.")
    # Keep lines before start_idx and after end_idx
    new_lines = lines[:start_idx] + lines[end_idx:]
else:
    print("WARNING: Line indices do not match expectations. Dumping context:")
    print(f"L{start_idx+1}: {lines[start_idx]}")
    print(f"L{end_idx}: {lines[end_idx-1]}")
    # Fallback: Search for the specific unique markers
    # Marker 1: </Dialog> (Line 707) -> We want to delete what follows until Available Rooms or </section>
    # Marker 2: </section > (Line 850)
    
    # Let's try to find the lines dynamically if indices are off
    try:
        dialog_close_idx = -1
        section_close_idx = -1
        
        for i, line in enumerate(lines):
            if "</Dialog>" in line:
                dialog_close_idx = i
            if "</section >" in line or (i > 700 and "</section>" in line and "offers-section" in lines[i+3]): 
                 # heuristic for line 850
                 if "</section >" in line:
                     section_close_idx = i
                     break
        
        if dialog_close_idx != -1 and section_close_idx != -1:
             start_idx = dialog_close_idx + 1
             end_idx = section_close_idx
             print(f"Dynamic adjusting: Deleting {start_idx+1} to {end_idx}")
             new_lines = lines[:start_idx] + lines[end_idx:]
        else:
             print("Could not locate block dynamically. Aborting deletion.")
             new_lines = lines
    except Exception as e:
        print(f"Error: {e}")
        new_lines = lines

# Perform Replacements in new_lines
final_lines = []
for line in new_lines:
    # 1. Fix </section > to </section>
    if "</section >" in line:
        line = line.replace("</section >", "      </section>")
        
    # 2. Update Special Offers link
    if "onClick={scrollToBooking}" in line:
        line = line.replace("onClick={scrollToBooking}", "onClick={() => setShowSearchModal(true)}")
        
    # 3. Update Promo Banner button
    if 'className="bg-white text-emerald-800 hover:bg-emerald-100"' in line and "onClick" not in line:
        line = line.replace('className="bg-white text-emerald-800 hover:bg-emerald-100"', 
                            'onClick={() => setShowSearchModal(true)} className="bg-white text-emerald-800 hover:bg-emerald-100"')
    
    final_lines.append(line)

with open(file_path, 'w', encoding='utf-8') as f:
    f.writelines(final_lines)

print("Home.js updated successfully.")

File: frontend/src/scripts/fix_home.js

const fs = require('fs');

const filePath = 'c:/S/Spensmer/frontend/src/pages/public/Home.js';
const content = fs.readFileSync(filePath, 'utf8');
let lines = content.split(/\r?\n/);

console.log(`Total lines: ${lines.length}`);

// Lines to delete: 708 to 849 (1-based) => indices 707 to 848
// Line 708 should be "</motion.div>"
// Line 849 should be "</motion.div>"

let startIdx = 707;
let endIdx = 849;

// Verification
const line708 = lines[startIdx] ? lines[startIdx].trim() : '';
const line849 = lines[endIdx - 1] ? lines[endIdx - 1].trim() : '';

console.log(`L708: ${line708}`);
console.log(`L849: ${line849}`);

let newLines = [];

if (line708 === '</motion.div>' && line849 === '</motion.div>') {
    console.log("Found exact block to delete.");
    newLines = lines.slice(0, startIdx).concat(lines.slice(endIdx));
} else {
    console.log("WARNING: Indices mismatch. Attempting dynamic find.");
    // Heuristic:
    // Find first </Dialog> around 707
    // Find </section > around 850
    let dialogIdx = -1;
    let sectionIdx = -1;

    for (let i = 0; i < lines.length; i++) {
        if (lines[i].includes('</Dialog>')) dialogIdx = i;
        if (lines[i].includes('</section >')) sectionIdx = i;
    }

    if (dialogIdx !== -1 && sectionIdx !== -1 && sectionIdx > dialogIdx) {
        startIdx = dialogIdx + 1; // Start deleting AFTER </Dialog>
        endIdx = sectionIdx; // Stop deleting BEFORE </section > (so keeping 850)
        console.log(`Dynamic range: Deleting indices ${startIdx} to ${endIdx - 1}`);
        newLines = lines.slice(0, startIdx).concat(lines.slice(endIdx));
    } else {
        console.log("Dynamic find failed. Aborting delete.");
        newLines = lines;
    }
}

// Replacements
const finalLines = newLines.map(line => {
    // 1. Fix </section >
    if (line.includes('</section >')) {
        return line.replace('</section >', '      </section>');
    }
    // 2. Update Special Offers link
    if (line.includes('onClick={scrollToBooking}')) {
        return line.replace('onClick={scrollToBooking}', 'onClick={() => setShowSearchModal(true)}');
    }
    // 3. Update Promo Banner button
    if (line.includes('className="bg-white text-emerald-800 hover:bg-emerald-100"') && !line.includes('onClick')) {
        return line.replace('className="bg-white text-emerald-800 hover:bg-emerald-100"',
            'onClick={() => setShowSearchModal(true)} className="bg-white text-emerald-800 hover:bg-emerald-100"');
    }
    return line;
});

fs.writeFileSync(filePath, finalLines.join('\n'), 'utf8');
console.log("Home.js updated.");
